homeassistant/components/miraie_ac/__init__.py

```diff
@@ -0,0 +1,46 @@
+"""The MirAIe AC integration."""
+from __future__ import annotations
+
+import logging
+
+from py_miraie_ac import AuthType, MirAIeAPI
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_PASSWORD, Platform
+from homeassistant.core import HomeAssistant
+
+from .const import CONFIG_KEY_USER_ID, DOMAIN
+
+PLATFORMS: list[Platform] = [Platform.CLIMATE]
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up MirAIe AC from a config entry."""
+
+    hass.data.setdefault(DOMAIN, {})
+    entry.async_on_unload(entry.add_update_listener(update_listener))
+
+    login_id = entry.data[CONFIG_KEY_USER_ID]
+    password = entry.data[CONF_PASSWORD]
+
+    api = MirAIeAPI(auth_type=AuthType.MOBILE, login_id=login_id, password=password)
+    await api.initialize()
```

Comment:
The library uses `paho_mqtt`, which is synced code and should not run in the event loop. Please run the code in an executor.
`api.initialize()` calls internally `broker.connect`, blocking the event loop until a connection is made.
Please run the broker connect in an executor.